components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/quotaLimit/ResourceQuotaLimitInterceptor.java

```diff
@@ -0,0 +1,116 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.apimgt.rest.api.util.interceptors.quotaLimit;
+
+import org.apache.cxf.common.util.UrlUtils;
+import org.apache.cxf.attachment.DelegatingInputStream;
+import org.apache.cxf.message.Message;
+import org.apache.cxf.phase.AbstractPhaseInterceptor;
+import org.apache.cxf.phase.Phase;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.quotaLimiter.ResourceQuotaLimiter;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.APIManagerConfiguration;
+import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+/**
+ * This class handles API creations considering specified quotas.
+ */
+public class ResourceQuotaLimitInterceptor extends AbstractPhaseInterceptor {
+
+    public ResourceQuotaLimitInterceptor() {
+        super(Phase.PRE_INVOKE);
+    }
+
+    /**
+     * Limits to create APIs if below conditions are satisfied.
+     * > ResourceQuotaLimit enabled
+     * > Return value as "true" from the getToBeQuotaLimited method.
+     *
+     * @param message cxf message
+     */
+    @Override
+    public void handleMessage(Message message) {
+        Message message1 = message;
+        try {
+            if (getQuotaLimitEnabled() && getToBeQuotaLimited(message) && getIsRegularAPICreation(message)) {
+                Map<String, String> queryParamsMap = UrlUtils.parseQueryString(
+                        (String) message.get(QuotaLimitInterceptorConstants.QUERY_PARAM_STRING));
+                if (queryParamsMap.containsKey(QuotaLimitInterceptorConstants.QUERY_PARAM_ORGANIZATION_ID)) {
+                    String organizationId = queryParamsMap.get(QuotaLimitInterceptorConstants.QUERY_PARAM_ORGANIZATION_ID);
+                    String userId = QuotaLimitInterceptorConstants.QUOTA_LIMIT_USERID;
+                    String resourceType = QuotaLimitInterceptorConstants.QUOTA_LIMIT_RESOURCE_TYPE;
+                    try {
+                        ResourceQuotaLimiter quotaLimiter = APIUtil.getResourceQuotaLimiter();
+                        if (quotaLimiter.getAPIQuotaLimitStatus(organizationId, userId, resourceType)) {
+                            Response response = Response.status(Response.Status.TOO_MANY_REQUESTS).build();
+                            message.getExchange().put(Response.class, response);
+                        }
+                    } catch (APIManagementException e) {
+                        e.printStackTrace();
```

Comment:
Shall we use logging instead of printing stack trace? Please apply to other places as well.